api/app/signals/apps/xperimental/rest_framework/filters/signal.py

```diff
@@ -0,0 +1,248 @@
+            'parent_signal': (Q(is_signal=False) & Q(is_parent=True) & Q(is_child=False)),
+            'exclude_parent_signal': ~(Q(is_parent=True) & Q(is_child=False)),
+            'child_signal': (Q(is_signal=False) & Q(is_parent=False) & Q(is_child=True)),
+        }
+
+        q_filter = q_objects[choices.pop()]
+        while choices:
+            q_filter |= q_objects[choices.pop()]
+
+        return queryset.filter(q_filter) if q_filter else queryset
+
+    def note_keyword_filter(self, queryset, name, value):
+        return queryset.filter(signal_id__in=Note.objects.filter(text__icontains=value).values_list('_signal_id'))
+
+    def punctuality_filter(self, queryset, name, value):
+        # When work on a Signal was finished, it can no longer be late and it is excluded.
+        queryset = queryset.exclude(status_state__in=[workflow.AFGEHANDELD, workflow.GEANNULEERD, workflow.GESPLITST])
+
+        # Historical data will not have deadlines calculated for it and can be
+        # filtered for by using this filter's "null" option.
+        if value == 'null':
+            return queryset.filter(category_assignment_deadline__isnull=True)
+
+        # Not interested in historical data without deadlines:
+        queryset = queryset.exclude(category_assignment_deadline__isnull=True)
+
+        local_now = now()
+        if value == 'on_time':
+            return queryset.filter(category_assignment_deadline__gt=local_now)
+        elif value == 'late':
+            return queryset.filter(category_assignment_deadline__lt=local_now)
+        elif value == 'late_factor_3':
+            return queryset.filter(category_assignment_deadline_factor_3__lt=local_now)
+
+    def directing_department_filter(self, queryset, name, value):
+        """
+        Filter Signals on directing department.
+
+        * A directing department can only be set on a Parent Signal
+        * When providing the option "null" select all parent Signals without one or more directing departments
+        * When providing on or more department codes as options select all parent Signals which match directing
+          departments
+
+        Example 1: "?directing_department=ASC" will select all parent Signals where ASC is the directing department
+        Example 2: "?directing_department=ASC&directing_department=null" will select all parent Signals without a
+                   directing department OR ASC as the directing department
+        Example 3: "?directing_department=null" will select all parent Signals without a directing department
+        """
+        choices = value  # we have a MultipleChoiceFilter ...
+        if len(choices) == len(department_choices()):
+            # No filters are present, or ... all filters are present. In that case we want all Signal instances
+            return queryset
+
+        # Directing departments are only set on parent Signals
+        parent_q_filter = (Q(is_parent=True) & Q(is_child=False))
```

Comment:
Why not just check `is_parent=True`?

Reply:
In the orignal filter we do `parent_q_filter = (Q(parent__isnull=True) & Q(children__isnull=False))` and it was copied from that. So I kept it like that just to be sure. It should not happen that both are True, but this is something we need to check on ACC/PROD